quantity test failures with numpy 1.9 and master

with current astropy master there are a bunch of quantity tests are failing with deprecation warnings:
e.g.

```
local/lib/python2.7/site-packages/astropy-1.0.dev9889-py2.7-linux-x86_64.egg/astropy/coordinates/sky_coordinate.py:199: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = [<Quantity 1000.0 0.001 karcsec>, <Latitude 2.0 karcsec>, <Quantity 3000.0 0.001 kpc>, <class 'astropy.coordinates.builtin_frames.Galactic'>]
kwargs = {'representation': <class 'astropy.coordinates.representation.SphericalRepresentation'>, 'unit': (Unit("karcsec"), Unit("karcsec"), Unit("kpc"))}

    def _get_frame(args, kwargs):
        """
        Determine the coordinate frame from input SkyCoord args and kwargs.  This
        modifies args and/or kwargs in-place to remove the item that provided
        `frame`.  It also infers the frame if an input coordinate was provided and
        checks for conflicts.

        This allows for frame to be specified as a string like 'icrs' or a frame
        class like ICRS, but not an instance ICRS() since the latter could have
        non-default representation attributes which would require a three-way merge.
        """
        frame = kwargs.pop('frame', None)

        if frame is not None:
            # Frame was provided as kwarg so validate and coerce into corresponding frame.
            frame_cls = _get_frame_class(frame)
            frame_specified_explicitly = True
        else:
            # Look for the frame in args
            for arg in args:
                try:
                    frame_cls = _get_frame_class(arg)
                    frame_specified_explicitly = True
                except ValueError:
                    pass
                else:
>                   args.remove(arg)
E                   DeprecationWarning: elementwise comparison failed; this will raise the error in the future.
```

additionally a failure due to a new ufunc in numpy master:

```
local/lib/python2.7/site-packages/astropy-1.0.dev9889-py2.7-linux-x86_64.egg/astropy/units/tests/test_quantity.py:935: DeprecationWarning
_________________________________ TestUfuncCoverage.test_coverage __________________________________

self = <astropy.units.tests.test_quantity_ufuncs.TestUfuncCoverage object at 0x7f2c6782d8d0>

    def test_coverage(self):
        all_np_ufuncs = set([ufunc for ufunc in np.core.umath.__dict__.values()
                             if type(ufunc) == np.ufunc])

        from .. import quantity_helper as qh

        all_q_ufuncs = (qh.UNSUPPORTED_UFUNCS |
                        set(qh.UFUNC_HELPERS.keys()))

>       assert all_np_ufuncs - all_q_ufuncs == set([])
E       assert set([<ufunc 'cbrt'>]) == set([])
E         Extra items in the left set:
E         <ufunc 'cbrt'>

```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -310,6 +310,9 @@ Other Changes and Additions
 
 - Ensure numpy 1.9 is supported. [#2917]
 
+- Ensure numpy master is supported, by making ``np.cbrt`` work with quantities.
+  [#2937] 
+
 0.4.1 (2014-08-08)
 ------------------
 

--- a/astropy/units/quantity_helper.py
+++ b/astropy/units/quantity_helper.py
@@ -3,6 +3,7 @@
 
 import numpy as np
 from .core import UnitsError, dimensionless_unscaled
+from ..utils.compat.fractions import Fraction
 
 
 def _d(unit):
@@ -64,6 +65,11 @@ UFUNC_HELPERS[np.square] = lambda f, unit: ([1.], unit ** 2 if unit is not None
 UFUNC_HELPERS[np.reciprocal] = lambda f, unit: ([1.], unit ** -1
                                                 if unit is not None
                                                 else dimensionless_unscaled)
+# cbrt only was added in numpy 1.10
+if isinstance(getattr(np, 'cbrt', None), np.ufunc):
+    UFUNC_HELPERS[np.cbrt] = lambda f, unit: ([1.], unit ** Fraction(1, 3)
+                                              if unit is not None
+                                              else dimensionless_unscaled)
 # ones_like was not private in numpy <= 1.6
 if isinstance(getattr(np.core.umath, 'ones_like', None), np.ufunc):
     UFUNC_HELPERS[np.core.umath.ones_like] = (lambda f, unit:

--- a/astropy/units/tests/test_quantity_ufuncs.py
+++ b/astropy/units/tests/test_quantity_ufuncs.py
@@ -264,6 +264,16 @@ class TestQuantityMathFuncs(object):
         assert np.all(np.reciprocal(np.array([1., 2., 4.]) * u.m)
                       == np.array([1., 0.5, 0.25]) / u.m)
 
+    # cbrt only introduced in numpy 1.10
+    @pytest.mark.skipif("not hasattr(np, 'cbrt')")
+    def test_cbrt_scalar(self):
+        assert np.cbrt(8. * u.m**3) == 2. * u.m
+
+    @pytest.mark.skipif("not hasattr(np, 'cbrt')")
+    def test_cbrt_array(self):
+        assert np.all(np.cbrt(np.array([1., 8., 64.]) * u.m**3)
+                      == np.array([1., 2., 4.]) * u.m)
+
     def test_power_scalar(self):
         assert np.power(4. * u.m, 2.) == 16. * u.m ** 2
         assert np.power(4., 200. * u.cm / u.m) == \

